src/main/java/org/datadog/jmxfetch/Status.java

```diff
@@ -87,15 +122,62 @@ private String generateYaml() {
         return yaml.dump(status);
     }
 
-    public void flush() {
+    private String generateJson() {
+        Gson gson = new Gson();
+        HashMap<String, Object> status = new HashMap<String, Object>();
+        status.put("timestamp", System.currentTimeMillis());
+        status.put("checks", this.instanceStats);
+        return gson.toJson(status);
+    }
+
+    private boolean postRequest(String body, int port) {
+        int responseCode = 0;
+        try {
+            String url = "https://localhost:" + port + "/agent/jmxstatus";
```

Comment:
We make the assumption in a lot of places in the new agent that `localhost` actually works, but we've experienced a lot of cases where `localhost` doesn't work properly (and users have to customize this to `127.0.0.1` or `::`). The Statsd reporter actually supports the configuration of the host it reports to because of this.

Not something we need to tackle now, but we should do it at some point :)